cluster/addons/iso-client/coreaffinity-isolator.go

```diff
@@ -24,37 +27,73 @@ const (
 	name = "iso"
 )
 
+func handleSIGTERM(sigterm chan os.Signal, client *aff.EventDispatcherClient, opaque *opaq.OpaqueIntegerResourceAdvertiser) {
+	<-sigterm
+	shutdownIsolator(client, opaque)
+}
+
+func shutdownIsolator(client *aff.EventDispatcherClient, opaque *opaq.OpaqueIntegerResourceAdvertiser) {
+	unregisterRequest := &lifecycle.UnregisterRequest{
+		Name:  client.Name,
+		Token: client.Token,
+	}
+
+	if _, err := client.Unregister(client.Ctx, unregisterRequest); err != nil {
+		opaque.RemoveOpaqueResource()
+		glog.Fatalf("Failed to unregister handler: %v")
+	}
+	glog.Infof("Unregistering custom-isolator: %s", name)
+
+	if err := opaque.RemoveOpaqueResource(); err != nil {
+		glog.Fatalf("Failed to remove opaque resources: %v", err)
+	}
+
+	os.Exit(0)
+
+}
+
 // TODO: split it to smaller functions
 func main() {
 	flag.Parse()
 	glog.Info("Starting ...")
-	cpuTopo, err := discovery.DiscoverTopology()
+	topology, err := discovery.DiscoverTopology()
 	if err != nil {
 		glog.Fatalf("Cannot retrive CPU topology: %q", err)
 	}
+	glog.Infof("Detected topology: %v", topology)
 
-	glog.Infof("Detected topology: %v", cpuTopo)
+	opaque, err := opaq.NewOpaqueIntegerResourceAdvertiser(name, fmt.Sprintf("%d", topology.GetTotalCPUs()))
+	if err != nil {
+		shutdownIsolator(nil, opaque)
+		glog.Fatalf("Cannot create opaque resource advertiser: %v", err)
+	}
+	if err = opaque.AdvertiseOpaqueResource(); err != nil {
+		shutdownIsolator(nil, opaque)
+		glog.Fatalf("Failed to advertise opaque resources: %v", err)
```

Comment:
Unreachable code (`shutdownIsolator` not expected to return) -- log at error level prior to `shutdownIsolator` call?